ibis/pandas/execution/tests/test_timecontext.py

```diff
@@ -0,0 +1,54 @@
+import pandas as pd
+import pandas.util.testing as tm
+import pytest
+
+import ibis
+
+pytestmark = pytest.mark.pandas
+
+
+def test_execute_with_timecontext(time_table):
+    expr = time_table
+    # define a time context for time-series data
+    context = ("20170101", "20170103")
+
+    # without time context, execute produces every row
+    df_all = expr.execute()
+    assert len(df_all["time"]) == 4
+
+    # with context set, execute produces only rows within context
+    df_within_context = expr.execute(timecontext=context)
+    assert len(df_within_context["time"]) == 1
+
+
+merge_asof_minversion = pytest.mark.skipif(
+    pd.__version__ < '0.19.2',
+    reason="at least pandas-0.19.2 required for merge_asof",
+)
+
+
+@merge_asof_minversion
+def test_context_adjustment_asof_join(
```

Comment:
Can you also add a rule for window operations and test the post execute trimming as well?

Reply:
Added in the same file.